Source/Scene/GroundPolygon.js

```diff
@@ -0,0 +1,424 @@
+        var ibIndices = IndexDatatype.createTypedArray(numVertices, numIndices);
+
+        var i0;
+        var i1;
+        var i2;
+
+        index = 0;
+        for (i = 0; i < numBottomIndices; i += 3) {
+            i0 = bottomIndices[i] * 2;
+            i1 = bottomIndices[i + 1] * 2;
+            i2 = bottomIndices[i + 2] * 2;
+
+            ibIndices[index++] = i2;
+            ibIndices[index++] = i1;
+            ibIndices[index++] = i0;
+        }
+
+        for (i = 0; i < numBottomIndices; i += 3) {
+            i0 = bottomIndices[i] * 2;
+            i1 = bottomIndices[i + 1] * 2;
+            i2 = bottomIndices[i + 2] * 2;
+
+            ibIndices[index++] = i0 + 1;
+            ibIndices[index++] = i1 + 1;
+            ibIndices[index++] = i2 + 1;
+        }
+
+        var offset = numCapVertices;
+        for (i = 0; i < numWalls; ++i) {
+            wall = walls[i];
+            var wallIndices = wall.indices;
+            wallLength = wallIndices.length;
+
+            for (j = 0; j < wallLength; ++j) {
+                ibIndices[index++] = wallIndices[j] + offset;
+            }
+            offset += wall.attributes.position.values.length / 3;
+        }
+
+        var positionBuffer = context.createVertexBuffer(vbPositions, BufferUsage.STATIC_DRAW);
+        var normalBuffer = context.createVertexBuffer(vbNormals, BufferUsage.STATIC_DRAW);
+
+        var indexDatatype = (ibIndices.BYTES_PER_ELEMENT === 2) ?  IndexDatatype.UNSIGNED_SHORT : IndexDatatype.UNSIGNED_INT;
+        var indexBuffer = context.createIndexBuffer(ibIndices, BufferUsage.STATIC_DRAW, indexDatatype);
+
+        var attributes = [{
+            index                  : attributeLocations.positionHigh,
+            vertexBuffer           : positionBuffer,
+            componentsPerAttribute : 3,
+            componentDatatype      : ComponentDatatype.FLOAT,
+            offsetInBytes          : 0,
+            strideInBytes          : ComponentDatatype.getSizeInBytes(ComponentDatatype.FLOAT) * 3 * 2
+        }, {
+            index                  : attributeLocations.positionLow,
+            vertexBuffer           : positionBuffer,
+            componentsPerAttribute : 3,
+            componentDatatype      : ComponentDatatype.FLOAT,
+            offsetInBytes          : ComponentDatatype.getSizeInBytes(ComponentDatatype.FLOAT) * 3,
+            strideInBytes          : ComponentDatatype.getSizeInBytes(ComponentDatatype.FLOAT) * 3 * 2
+        }, {
+            index                  : attributeLocations.normal,
+            vertexBuffer           : normalBuffer,
+            componentsPerAttribute : 3,
+            componentDatatype      : ComponentDatatype.FLOAT
+        }];
+
+        polygon._va = context.createVertexArray(attributes, indexBuffer);
+
+        polygon._boundingSphere = BoundingSphere.fromPoints(outerRing);
+        polygon._boundingSphere.radius = upDelta; // TODO: Correct bounding sphere
```

Comment:
Yes, please.  Recall [this](https://github.com/AnalyticalGraphicsInc/cesium/issues/2172#issuecomment-74669808).